clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -204,23 +204,14 @@ public synchronized void ensureCoordinatorReady() {
                 time.sleep(retryBackoffMs);
             }
         }
+        // Since coordinator is set in a synchronized block in the group coordinator response callback,
+        // the future associated with the cooordinator request can be safely cleared here.
+        findCoordinatorFuture = null;
```

Comment:
While looking at the HeartbeatThread, it has the following line, which shadows the `findCoordinatorFuture` class field:

```java
                RequestFuture findCoordinatorFuture = null;
```

@hachikuji, is it intentional to have that there?